lib/problems.rb

```diff
@@ -3,7 +3,32 @@
 # Time Complexity: ?
 # Space Complexity: ?
 def balanced(string)
-  raise NotImplementedError, "Not implemented yet"
+  array = string.split('')
+  return false if array.length % 2 != 0
+
+  stack = Stack.new
+  array.each do |ele|
+    stack.push(ele)
+  end
+
+  array.length.times do |i|
+    remove = stack.pop
+    case remove
+    when '['
+      return false if array[i] != ']'
+    when ']'
+      return false if array[i] != '['
+    when '{'
+      return false if array[i] != '}'
+    when '}'
+      return false if array[i] != '{'
+    when '('
+      return false if array[i] != ')'
+    when ')'
+      return false if array[i] != '('
```

Comment:
Just a note that you can dry this up with a hash where they keys are the close braces and the values are the corresponding open braces.

```
return false if brace_hash[remove] != array.pop
```